src/pushsource/_impl/model/base.py

```diff
@@ -248,3 +249,19 @@ def with_checksums(self):
             updated_sums[attribute] = hasher.hexdigest()
 
         return attr.evolve(self, **updated_sums)
+
+    @property
+    def content(self):
```

Comment:
I'm not sure this should be a property, I think a plain method fits better.

It will work either way, but I think developers usually have an expectation that properties are cheap, and not subject to failure.  It doesn't fit this case well, as this could potentially have to do HTTP requests or other slow and error-prone operations.

It also returns a different object on each call, which is not how a property would usually work.  For instance someone might write code like this, intending to read the content in chunks:

```python
while chunk := item.content.read():
  # oops, this is an infinite loop because every iteration
  # opens a new file and reads from the beginning!
```

This code is wrong but it doesn't *look* wrong.  I think the above would more clearly stand out as wrong if it were a method rather than a property.

Reply:
Thanks for pointing this. `FileContentReader` uses the same file object, however I missed that this created a new `FileContentReader` every time.